Fix Cart rule restriction 8.2.x

Sollsy opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.2.x
| Description?      | Fixes issue with cart rule restrictions. After changes works similarly as specific price rules where product must match all rules from rule group instead of just one but allows you to make more specific cart rule restrictions. tested on 1.7.8.x / 8.2.x
| Type?             | bug fix
| Category?         | FO, BO 
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | create similar cart rules as on image below
| UI Tests          |  CI test are green
| Fixed issue or discussion?     | Fixes #15525, #26528, #38078 
| Related PRs       |  
| Sponsor company   |  

1.7.8.x
![cartrule](https://github.com/user-attachments/assets/1889a20d-bd78-4400-b6d9-2936244f4e1d)

8.2.x

![cartrule8 2](https://github.com/user-attachments/assets/f9d9bfb1-05f2-4370-8670-ca1bcbea36bb)

This is my first time using github. I apologize if i put this inside wrong branch.

This is know issue with cart rules since forever. With these changes you can make more restrictive cart rules.

Display product rules group similar to specific price rules. It's easier understand how rule is going to work:
<img width="2336" height="1202" alt="image" src="https://github.com/user-attachments/assets/6b9656e6-84a9-4aaf-bb03-ebfbdfe5be89" />

Sollsy committed:
Update CartRule.php

Sollsy committed:
Update product_rule_group.tpl

Sollsy committed:
Update conditions.tpl

Sollsy committed:
Update CartRule.php

Sollsy committed:
Update CartRule.php

Sollsy committed:
Update CartRule.php

Sollsy committed:
Update CartRule.php

ps-jarvis commented:
<!-- PR_WELCOME -->
Hello @Sollsy!

This is your first pull request on [PrestaShop](https://github.com/PrestaShop/PrestaShop/) repository of the PrestaShop project.

Thank you, and welcome to this Open Source community!

Sollsy committed:
Update CartRule.php

Sollsy committed:
Allow discout on gift product variant CartRule.php

When you add gift to the cart user can increase quantity or add product variant. By default gift product is excluded from cart rules.

checkGiftCount allows you to check gift quantity in cart and allow discount calculation for quantity of gift > 1.

Test: make cart rule with product restriction and discount with add gift enabled. Add another variant of selected gift  for example different size to cart and check discounts.

Sollsy committed:
missing checkGiftCount query

Added missing sql query for checkGiftCount.

Sollsy committed:
Update CartRule.php

Sollsy committed:
Update CartRule.php

Sollsy committed:
display different group conditions.tpl

Sollsy committed:
indicate each product_rule_group with panel class

Display different product rule groups with numbers indicating that it has to match them. Similar to specific price rules.

Sollsy committed:
fix price calculation on cheapest product gift

Fix price calculation when adding product and discount on cheapest. By default discount is calculated from gift price.

After changes checkGiftCount function properly checks for cheapest product. If user adds quantity to the gift product, it will recalculate discount since gift product quantity is greater than 1.

Sollsy committed:
fix price calculation on cheapest product gift

fix gift query

Sollsy committed:
fix gift chain with autoApply

Fixed multiple gift chains with autoApply rules where gift with quantity 1 could add another gift if restrictions were met. After changes it keeps count of product and gifts.

Sollsy committed:
Update CartRule.php

Sollsy committed:
fix condition count CartRule.php

Sollsy committed:
Update CartRule.php
